### ISSUE TITLE
Tweet replies don't get sent as replies when saved as drafts

### ISSUE THREAD

**Issue by Yowlen** [TwidereProject/Twidere-Android#884]
## Steps to reproduce
Essentially the title. If a reply gets saved as a draft, such as with a timeout from a network error, the information about where it's replying to doesn't get used when sending it - even though it still properly shows as a reply when the draft is opened.

## Extra info

Android version: 5.1.1

Network type:

- [X] Cellular
- [X] Wi-Fi

App version: I think it's been present since 3.4.x, maybe earlier, but definitely still present in 3.6.21

Build variant:

- [X] Google
- [ ] F-Droid

Micro-blogging service:

- [X] Twitter
- [ ] Fanfou
- [ ] GNU Social / StatusNet
- [ ] Mastodon

App settings:

- [ ] Custom API
- [ ] Custom consumer key/secrets
- [X] Official keys
- [ ] Using [Thumbor](https://github.com/thumbor/thumbor)
- [ ] Connected through proxy
- [X] Media preview enabled

**Comment by igalic** [comment]
i've notice this too, but with the following … let's call it workflow:

- send as reply tweet
- tweet gets saved as draft because network drop
- send reply tweet again from draft

tweet is now posted as both, reply and standalone tweet because the network drop only made us miss the response code, so we assumed it wasn't posted

**Comment by Yowlen** [comment]
Yes, that has happened to me too. I forgot to include it in the main report.

I've since gotten a new router, so the incoming packet loss shouldn't be an issue for me anymore, but it's still good to have Twidere implement checks to keep the tweet chain intact, as well as make sure the tweet doesn't get double-posted.

But yeah, cheap routers suck. 😆
